raiden_contracts/data/source/raiden/TokenNetwork.sol

```diff
@@ -263,6 +272,7 @@ contract TokenNetwork is Utils {
         settlement_timeout = _settlement_timeout_min;
         settlement_timeout_min = _settlement_timeout_min;
         settlement_timeout_max = _settlement_timeout_max;
+        claim_signer = _claim_signer;
```

Comment:
IMO this should address should be recovered from the token contract. My rationale is that this functionality must be supported by the token anyways, getting the address directly from the token will prevent deployment bugs and would also allow a token to change the claim address if it wants. (For that I would just use a function like `is_valid_signer` in the token, the implementation in the token contract would then be undefined, this does lead to some hard edge cases on gas estimation though, but I guess it is the price to pay for the feature)

Reply:
note: @karlb argues that since we want to have a more efficient version of the code, which skips the steps of claiming, approving, and then transferring, and directly burns and transfers the tokens. For that optimized path the TN needs to be trusted by the Token, if that is the case, then @karlb argues that this approach is also fine.